aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/HttpRequestChecksumDecorator.kt

```diff
@@ -148,9 +193,111 @@ class HttpRequestChecksumCustomization(
+            return emptySection
+        }
+
+        // Otherwise we write the necessary sections to the service config
+        return when (section) {
+            is ServiceConfig.ConfigImpl ->
+                writable {
+                    rustTemplate(
+                        """
+                        /// Return a reference to the request_checksum_calculation value contained in this config, if any.
+                        pub fn request_checksum_calculation(&self) -> #{Option}<&#{RequestChecksumCalculation}> {
+                            self.config.load::<#{RequestChecksumCalculation}>()
+                        }
+                        """,
+                        *codegenScope,
+                    )
+                }
+
+            is ServiceConfig.BuilderImpl ->
+                writable {
+                    rustTemplate(
+                        """
+                        /// Set the [`RequestChecksumCalculation`](#{RequestChecksumCalculation})
+                        /// to determine when a checksum will be calculated for request payloads.
+                        pub fn request_checksum_calculation(
+                            mut self,
+                            request_checksum_calculation: #{RequestChecksumCalculation}
+                        ) -> Self {
+                            self.set_request_checksum_calculation(#{Some}(request_checksum_calculation));
+                            self
+                        }
+                        """,
+                        *codegenScope,
+                    )
+
+                    rustTemplate(
+                        """
+                        /// Set the [`RequestChecksumCalculation`](#{RequestChecksumCalculation})
+                        /// to determine when a checksum will be calculated for request payloads.
+                        pub fn set_request_checksum_calculation(
+                            &mut self,
+                            request_checksum_calculation: #{Option}<#{RequestChecksumCalculation}>
+                        ) -> &mut Self {
+                            self.config.store_or_unset(request_checksum_calculation);
+                            self
+                        }
+                        """,
+                        *codegenScope,
+                    )
+                }
+
+            is ServiceConfig.BuilderFromConfigBag ->
+                writable {
+                    rustTemplate(
+                        "${section.builder}.set_request_checksum_calculation(${section.configBag}.load::<#{RequestChecksumCalculation}>().cloned());",
+                        *codegenScope,
+                    )
+                }
+
+            else -> emptySection
+        }
+    }
+}
+
+/**
+ * Determine if the current service contains any operations with the HttpChecksum trait
+ */
+fun serviceHasHttpChecksumOperation(codegenContext: ClientCodegenContext) =
+    codegenContext.serviceShape.allOperations
```

Comment:
**correctness**: You usually want to leverage one of the knowledge indexes to ensure you traverse the model correctly. IIRC this would travers literal operation shapes and miss resources defined on the model. In this case use `TopDownIndex` most likely.

Reply:
Updated to use knowledge index in https://github.com/smithy-lang/smithy-rs/pull/3845/commits/618240137f2b03f7bd7a068693390bc0c487a322